pkg/client/apiutil/filterdiscoveryclient.go

```diff
@@ -0,0 +1,104 @@
+package apiutil
+
+import (
+	openapi_v2 "github.com/googleapis/gnostic/OpenAPIv2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/version"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+)
+
+// defaultRetries is the number of times a resource discovery is repeated if an api group disappears on the fly (e.g. ThirdPartyResources).
+const defaultRetries = 2
+
+type FilterDiscoveryClient struct {
```

Comment:
Needs godoc

Reply:
should this be internal?

Reply:
Added godoc. I personally prefer to keep it public because it is a generic wrapper to discovery interface, which might be useful in a different manner if someone wants to set a custom mapper.